### ISSUE TITLE
Plan for a 1.0 release

### ISSUE THREAD

**Issue by valberg** [percipient/django-allauth-2fa#119]
I was thinking that we should have a plan for a 1.0 release where we define:

- a deprecation policy
- a policy for which versions of dependency to support
- a versioning scheme (semver is fine, but we should be explicit about what we use)

**Comment by akx** [comment]
Should we do one more <1.0 release (0.9?) that

* has changes since 0.8 (https://github.com/valohai/django-allauth-2fa/compare/0.8...main)
* supports the versions of Django we currently test on

and then drop testing for [EOL versions of Django](https://endoflife.date/django) for the forthcoming 1.0?

**Comment by clokep** [comment]
Please shout if you need help doing the release! I think I dropped what I used to do into an email, but to put it in public:

> Unfortunately I don't think we have a releasing doc, but the steps I would follow were something like:
>
> 1. Ensure the release notes were up-to-date (and double check the date on it).
> 1. Update the version number in setup.py
> 1. Commit
> 1. Tag
> 1. git clean -dxf
> 1. python setup.py bdist_wheel sdist
> 1. twine check dist/*
> 1. twine upload dist/*

Some of the ecosystem has moved on since then, but I think the above should still work OK. 👍 

Other things to check are that access to ReadTheDocs works properly.

It might make sense to do a 0.9 just to ensure all the processes work OK. 🤷

**Comment by valberg** [comment]
> Should we do one more <1.0 release (0.9?) that
> 
>     * has changes since 0.8 ([0.8...main](https://github.com/valohai/django-allauth-2fa/compare/0.8...main))
> 
>     * supports the versions of Django we currently test on
> 
> 
> and then drop testing for [EOL versions of Django](https://endoflife.date/django) for the forthcoming 1.0?

Yeah I think that is a very good idea! My thoughts were more in the planning department, ie. what do we want to have before we can release a version 1.0 :)

**Comment by monosans** [comment]
How about an intermediate release of v0.11 now? A lot of changes have accumulated that would be nice to have.

**Comment by jkaeske** [comment]
> How about an intermediate release of v0.11 now? A lot of changes have accumulated that would be nice to have.

Any news on this release?

**Comment by federicobond** [comment]
Hi @akx, is there any way I can help push a new release of the library out?

**Comment by akx** [comment]
@monosans @jkaeske @federicobond @matmair @SchrodingersGat 

0.11.0 released (with the fancy new Trusted Publisher system too) 🎉  

* PyPI: https://pypi.org/project/django-allauth-2fa/0.11.0/
* GH: https://github.com/valohai/django-allauth-2fa/releases/tag/v0.11.0

**Comment by matmair** [comment]
You rock @akx @valberg !
